src/osd/ReplicatedPG.cc

```diff
@@ -5458,6 +5465,7 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 
 	write_update_size_and_usage(ctx->delta_stats, oi, ctx->modified_ranges,
 				    op.clonerange.offset, op.clonerange.length, false);
+	ctx->clean_regions.mark_data_region_dirty(op.clonerange.offset, op.clonerange.length);
```

Comment:
If ctx->modified_ranges isn't correct for those cases, we have a larger problem since we use it for recovery...  Yeah, you should definitely fix those cases and use ctx->modified_ranges.